### PR TITLE
Status system

### PR DESCRIPTION
**Pull request by ImUrX**
This will let the server notify of problems that in the future might be autofixable, but for now just will pass to the GUI and show them as a problem requiring fixing.

Requires #647 and SlimeVR/SolarXR-Protocol#110

### CODE REVIEW

**Comment by ImUrX** [comment]
Fixes #653 and partially #655

File: server/src/main/java/dev/slimevr/status/StatusSystem.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		listeners.add(listener)
+	}
+
+	fun removeListener(listener: StatusListener) {
+		listeners.remove(listener)
+	}
+
+	fun getStatuses(): Array<StatusMessageT> = statuses.map { (id, message) ->
+		val status = StatusMessageT()
+		status.id = id.toUInt().toLong()
```
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/status/StatusSystem.kt:26]: I don't quite get this lol
```suggestion
		status.id = id.toLong()
```

File: server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt [hunk 2]
```diff
@@ -52,13 +58,44 @@ class Tracker @JvmOverloads constructor(
...
+			if (field == value) return
+			field = value
+
+			if (!isInternal) {
+				// Check if a full reset is needed
+				if (needsReset && value != null && lastResetStatus == 0u && status.sendData) {
+					reportRequireReset()
+				} else if (value == null && lastResetStatus != 0u) {
+					vrServer.statusSystem.removeStatus(lastResetStatus)
+					lastResetStatus = 0u
```
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt:98]: This seems like an incredibly large amount of logic to have in a property setter...

File: server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt [hunk 3]
```diff
@@ -84,6 +121,53 @@ class Tracker @JvmOverloads constructor(
...
+				}
+				trackerNum = tempTrackerNum
+			}
+		}
+		val status = StatusDataUnion().apply {
+			type = StatusData.StatusTrackerError
+			value = statusMsg
+		}
+		lastErrorStatus = vrServer.statusSystem.addStatus(status, true)
+	}
```
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt:169]: This feels like a little awkward of a place to have these?

File: server/src/main/java/dev/slimevr/status/StatusSystem.kt (hunk 1 above)
Comment (ImUrX) [reply server/src/main/java/dev/slimevr/status/StatusSystem.kt:26]: ``-52.toLong()`` would just be ``-52``, ``-52.toUInt()`` would convert it into ``4294967244``

File: server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt (hunk 2 above)
Comment (ImUrX) [reply server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt:98]: any suggestions?

File: server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt (hunk 3 above)
Comment (ImUrX) [reply server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt:169]: the problem is the variable used to hold the latest status reported by the own tracker

File: server/src/main/java/dev/slimevr/status/StatusSystem.kt (hunk 1 above)
Comment (Louka3000) [reply server/src/main/java/dev/slimevr/status/StatusSystem.kt:26]: Why would you not want it to go in the negatives? For solarxr?

Comment (ImUrX) [reply server/src/main/java/dev/slimevr/status/StatusSystem.kt:26]: We are in Kotlin, no purpose of being forced to use `int`s. But I'm still kind of forced to >~>

**Review by ButterscotchV — APPROVED** [review]

**Review by Eirenliel — APPROVED** [review]

**Comment by ImUrX** [comment]
Ready for merge ^-^

**Pull request merged**